code/controllers/subsystem/SSticker.dm

```diff
@@ -227,9 +229,13 @@ SUBSYSTEM_DEF(ticker)
 			P.ready = FALSE
 
 	//Configure mode and assign player to special mode stuff
-	mode.pre_pre_setup()
+
 	var/can_continue = FALSE
 	can_continue = mode.pre_setup() //Setup special modes. This also does the antag fishing checks.
+	if(can_continue == LATE_HANDOUT)
+		var/random_time = rand(300 SECONDS, 900 SECONDS)
+		timer_till_handout = addtimer(CALLBACK(src, PROC_REF(handle_late_handout)), random_time)
```

Comment:
this is killing me, this would be better as just putting timers on the traitor datums individually (and you would also have people getting objectives at different times, which is cool)

Reply:
IIRC there was something against it being individual times, but I'll at least move the timer to go on the gamemode so it won't go on the SSticker

Reply:
Moved to be on the gamemode, Charlie made a decent point that you can be fucked if 2 traitors get the objective to steal the same item, and one gets it 10 minutes before the other, it can cause frustration

Reply:
IMO, I dont think thats an issue for traitors, they have advanced pinpointers, even better if they fight for it.